### ISSUE TITLE
emulator raises "GET" ... "Invalid request, unable to process." on first function call

### ISSUE THREAD

**Issue by ekalosak** [firebase/firebase-functions-python#130]
This error pops up every time I call a function after starting the emulators.
```
... [call stack omitted]                                                                                                     
>  Message: 'Request has invalid method.'                                               
>  Arguments: ('GET',)                                                                                                                                                          
>  ERROR:root:Invalid request, unable to process.
```
I'm using an `@https_fn.on_call` and calling it with `curl -X POST` so the `GET` in there is a bit confusing. Moreover, this just shows up - it doesn't degrade functionality much aside from potentially delaying the cold-start.

The functions work just fine... but this long stack trace on a cold-start is troubling.

**Comment by exaby73** [comment]
Hello @ekalosak. Could you provide a minimal reproducible example as well as your python version and contents of your `requirements.txt`?

**Comment by berengamble** [comment]
I just put this in my functions when I see that:

```
    if req.method == "GET" and req.path == "/__/health":
        logger.debug("Returning OK to healthcheck")
        return https_fn.Response("OK")
```

**Comment by exaby73** [comment]
@ekalosak Sorry if I misunderstood the issue, but does https://github.com/firebase/firebase-functions-python/pull/134 fix this issue? Seems unrelated to me.

If indeed the issue is this extra GET request triggering the warning log with the `None` request body, we should probably either try to fix that or if it's not something we can control, at least document it. The PR is still valid, just not sure if this issue should be considered fixed once that's merged in

**Comment by ekalosak** [comment]
@exaby73 That strikes me as correct.

The next line above in the stack trace is:
```
/.../venv/lib/python3.11/site-packages/firebase_functions/private/util.py", line 125, in _on_call_valid_
method
>      _logging.warning("Request has invalid method.", request.method)
```

While I believe you are quite right that #134 will not fix the reported error, I believe the fix is something quite similar. The offending line is [in the same util.py](https://github.com/firebase/firebase-functions-python/blob/main/src/firebase_functions/private/util.py#L136).

## minimal reproducible example
`python -c 'import logging; logging.warning("won't work", ("GET",))'`

### expected result
```
--- Logging error ---
Traceback (most recent call last):
  File "/.../3.11.4/lib/python3.11/logging/__init__.py", line 1110, in emit
    msg = self.format(record)
          ^^^^^^^^^^^^^^^^^^^
  File "/.../3.11.4/lib/python3.11/logging/__init__.py", line 953, in format
    return fmt.format(record)
           ^^^^^^^^^^^^^^^^^^
  File "/.../3.11.4/lib/python3.11/logging/__init__.py", line 687, in format
    record.message = record.getMessage()
                     ^^^^^^^^^^^^^^^^^^^
  File "/.../3.11.4/lib/python3.11/logging/__init__.py", line 377, in getMessage
    msg = msg % self.args
          ~~~~^~~~~~~~~~~
TypeError: not all arguments converted during string formatting
Call stack:
  File "<string>", line 1, in <module>
Message: 'asdf'
Arguments: (('GET',),)
```

Notice that string coercion fails for tuples.

So that's an easy fix of course, just extract the string value from the 1-tuple `request.method` and 🚀

**Comment by ekalosak** [comment]
Fix: https://github.com/firebase/firebase-functions-python/pull/138

**Comment by ekalosak** [comment]
Any updates?

**Comment by ekalosak** [comment]
Fixed by merge of #138 .
